util/pkg/vfs/s3fs.go

```diff
@@ -587,40 +594,66 @@ func (p *S3Path) RenderTerraform(w *terraformWriter.TerraformWriter, name string
 		return fmt.Errorf("reading data: %v", err)
 	}
 
-	bucketDetails, err := p.getBucketDetails(ctx)
-	if err != nil {
-		return err
-	}
+	// render DO's terraform
+	if p.scheme == "do" {
 
-	tfProviderArguments := map[string]string{
-		"region": bucketDetails.region,
-	}
-	w.EnsureTerraformProvider("aws", tfProviderArguments)
+		content, err := w.AddFileBytes("digitalocean_spaces_bucket_object", name, "content", bytes, false)
+		if err != nil {
+			return fmt.Errorf("rendering DO file: %v", err)
+		}
 
-	content, err := w.AddFileBytes("aws_s3_object", name, "content", bytes, false)
-	if err != nil {
-		return fmt.Errorf("rendering S3 file: %v", err)
-	}
+		// retrieve space region from endpoint
+		endpoint := os.Getenv("S3_ENDPOINT")
+		if endpoint == "" {
+			return fmt.Errorf("S3 Endpoint is empty")
+		}
+		region := strings.Split(endpoint, ".")[0]
```

Comment:
Do we always expect a dot separator? If so, should we return an error if it's missing?

Reply:
Per DO documentation, it is always in the format `{region}.digitaloceanspaces.com` and if user attempts to specify an incorrect endpoint, the region will contain the entire string from `S3_ENDPOINT`. I think that shouldn't be a problem since eventually the terraform resource will not be created. So I am keeping it as-is unless you think we need that check.